Adding traffic-mirroring in Vagrant and Linux dev-environment (#62)
l3af-project/l3af-arch: dev_environment/README.md

  on which eBPF programs are running and how they are configured).  If the rate limiter eBPF program is loaded, this command should output a latency distribution histogram that
  is more distributed.<p align="center"><img src="https://user-images.githubusercontent.com/106849610/179867246-406a2841-5a49-4102-b42c-e9cbf07ce2c3.png" width="400" height="200"/></p>
* To see the eBPF program metrics, browse to `http://localhost:33000` on the
  host and log in to Grafana with the default username and password of `admin`.
  After logging in you will be able to view the preconfigured dashboards.

> * Traffic-mirroring:

Reviewer: ```suggestion

* Additional Steps for testing out traffic mirroring:
```

  1. Set `traffic_mirroring: 'true'` in [config.yaml](config.yaml)
  2. Redeploy the Vagrant script (`vagrant reload --provision`) to reflect new changes, such as creation of a GUE tunnel and an additional VM (Collector)
  3. Start traffic mirrroing via `curl -X POST http://localhost:37080/l3af/configs/v1/add -d "@cfg/traffic_mirroring_payload.json"` from the host
  4. Delete the default route by executing this command (`sudo ip r del 192.168.10.50 via 192.168.10.1 dev enp0s8`) on l3af-VM as it is not required in the current vagrant environment

>   5. SSH into Collector VM via `vagrant ssh collector` command and execute `sudo tcpdump -i enp0s8` to see the mirrored-GUE packets and `sudo tcpdump -i gue1` to see the mirrored-original packets when we send traffic to the l3af-VM's web server (`hey -n 200 -c 20 http://localhost:18080`) from the host

Reviewer: ```suggestion
  5. SSH into Collector VM via `vagrant ssh collector` command and execute `sudo tcpdump -i enp0s8` to see the mirrored-GUE packets and `sudo tcpdump -i gue1` to see the mirrored-original packets when we send traffic to the l3af VM web server (`hey -n 200 -c 20 http://localhost:18080`) from the host
```

* To see the eBPF program metrics, browse to `http://localhost:33000` on the

>   host and login to Grafana with the default username and password of `admin`.

Reviewer 2: ```suggestion
  host and log in to Grafana with the default username and password of `admin`.
```

Author: Fixed in the newer commit

  After logging in you will be able to view the preconfigured dashboards.
* To see the eBPF program metrics, browse to `http://localhost:33000` on the
  host and login to Grafana with the default username and password of `admin`.

>   After logging in you will be able to view the preconfigured dashboards.

Reviewer 2: Duplicate of lines 69-71

Author: Fixed in the newer commit

* Traffic-mirroring:
  1. Set `traffic_mirroring: 'true'` in [config.yaml](config.yaml)
  2. Redeploy the Vagrant script (`vagrant reload --provision`) to reflect new changes, such as creation of a GUE tunnel and an additional VM (Collector)

>   3. Start traffic mirrroing via `curl -X POST http://localhost:37080/l3af/configs/v1/update -d "@cfg/traffic_mirroring_payload.json"` from the host

Author: Fixed in the newer commit
